nucliadb_texts/src/writer.rs

```diff
@@ -225,35 +231,31 @@ impl TextWriterService {
             .as_ref()
             .expect("Missing resource created date in metadata");
 
-        let mut doc = doc!(
-            self.schema.uuid => resource.resource.as_ref().expect("Missing resource details").uuid.as_str(),
+        let mut base_doc = doc!(
+            self.schema.uuid => resource_id,
             self.schema.modified => timestamp_to_datetime_utc(modified),
             self.schema.created => timestamp_to_datetime_utc(created),
             self.schema.status => resource.status as u64,
         );
 
-        #[allow(clippy::iter_cloned_collect)]
-        let resource_labels: Vec<String> = resource.labels.iter().cloned().collect();
-        for label in resource_labels {
+        for label in resource.labels.iter() {
             let facet = Facet::from(label.as_str());
-            doc.add_facet(self.schema.facets, facet);
+            base_doc.add_facet(self.schema.facets, facet);
         }
 
         for (field, text_info) in &resource.texts {
-            let mut subdoc = doc.clone();
+            let mut doc = base_doc.clone();
```

Comment:
Maybe call it `new_doc` instead of just `doc` to not create confusion with the macro.

Reply:
What about `field_doc`, as it's a doc for a field?

Reply:
sounds good